whois-api/src/main/java/net/ripe/db/whois/api/apiKey/BearerTokenExtractor.java

```diff
@@ -56,9 +60,12 @@ private OAuthSession getOAuthSession(final String bearerToken, final String apiK
               LOGGER.debug("JWT signature verification failed for {}", apiKeyId);
               return new OAuthSession(apiKeyId);
             }
-            
+
             //TODO[MA]: remove when apiKeyId is available from api registry call
-            return OAuthSession.from(new ObjectMapper().readValue(signedJWT.getPayload().toString(), OAuthSession.class), apiKeyId);
+            final OAuthSession oAuthSession = OAuthSession.from(new ObjectMapper().readValue(signedJWT.getPayload().toString(), OAuthSession.class), apiKeyId);
+
+            return ApiKeyUtils.validateAudience(oAuthSession, whoisKeycloakId) ? oAuthSession : new OAuthSession(apiKeyId);
```

Comment:
Will authentication silently fail if a `new OAuthSession()` is returned? If so can you add logging, e.g. `client id X didn't match Y` for example

Reply:
As INFO? or WARN?

Reply:
I was thinking that but we do allow user to use multiple type of authentication and also I would prefer it goes through our update flow

Reply:
OK. Is it possible to add a warning to the update and return to the user in the ack if the api key is invalid ?

Reply:
I think so, It should be possible. I will check

Reply:
Agreed to use message 
```The API key cannot be used because it was created for a different application or environment```